### ISSUE TITLE
Add CPT of the geological time scale

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#4448]
Add the CPT with the colors defined by the ICS.

https://ccgm.org/img/cms/2013%20Chart%20CGMW%20color%20codes.pdf

There are some done by Christian Heine. Maybe only the times must be checked. 
https://tectonicwaters.wordpress.com/2014/09/06/geological-time-scale-color-palettes-for-gmt/
https://bitbucket.org/chhei/gmt-cpts/src/master/
http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/

**Are you willing to help implement and maintain this feature?** Yes

**Comment by Esteban82** [comment]
Maybe this could be discussed in the next Thursday in the community meeting.

**Comment by PaulWessel** [comment]
Yes, I will add it to the agenda

**Comment by Esteban82** [comment]
I copy the notes from the community meetings

-  Can be tricky to add too many domain specific ones (which ones do we add and which ones we don’t?)
- Can have them managed as remote data files
- CPTCity is great for this already
- Have a `makecpt -C @somecpt`? Point to some cpt-city URL.
-  Need to be careful with license and citations if we’re storing CPTs

In general, think it is OK to add to cache but not to GMT core since it is not generic and not tied to data we provide
- What do we do with the data cache? There are a lot of datasets but they are not listed
- We don’t want to have to maintain a lot of data files
- Could have a clearer naming scheme (like @sample_* for every dataset used in the docs).-

**Comment by Esteban82** [comment]
I shared some plots with the geological CPT. I think that the plots can be useful. Maybe one or two of these plots can be polished and add in the gallery (or in the forum showcase).

Batchs scripts: [Earth_Age_CPT-City.zip](https://github.com/GenericMappingTools/gmt/files/5578893/Earth_Age_CPT-City.zip)

The important commands used were (in a batch file): 

```	
REM Links of the CPT
REM Geological CPT
rem	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GTS2012_eons.cpt"
rem	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GTS2012_eras.cpt"
rem	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GTS2012_periods.cpt"
	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GTS2012_epochs.cpt"
rem	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GTS2012_ages.cpt"

REM Gee Kent 2007
	gmt which -G > %color% "http://soliton.vm.bytemark.co.uk/pub/cpt-city/heine/GeeK07.cpt"

REM Only for DOS
        set /p cpt=<%color%

REM Plot
        gmt grdgradient @earth_relief_05m_p -A270 -G%SHADOW% -Ne0.5
	gmt grdimage -R -J -O -K @earth_age_05m_p >> %OUT% -C%cpt% -I%SHADOW%
	gmt psscale -R -J -O -K -DJRM+o0.3c/0+w-7/0.618c -C%cpt% -I >> %OUT% -G0/200 -L0 
rem	gmt psscale -R -J -O -K -DJRM+o0.3c/0+w-7/0.618c -C%cpt% -I >> %OUT% -G0/200 -B+l"Age (Ma)" 
	gmt psbasemap -R -J -O -K >> %OUT% -B0
	gmt pscoast -R -J -O -K  >> %OUT% -Df -W1/faint
```



![Earth_Age_CPT-City_Periods_Ma](https://user-images.githubusercontent.com/17908951/99893410-b6c76180-2c5e-11eb-89a1-94f67a4a7e6f.png)


![Earth_Age_CPT-City_Periods_Names](https://user-images.githubusercontent.com/17908951/99893411-b929bb80-2c5e-11eb-92c8-926b525988e6.png)


![Earth_Age_CPT-City_Epochs_Ma](https://user-images.githubusercontent.com/17908951/99893413-bd55d900-2c5e-11eb-9fe7-8d8f0d83fadd.png)
![Earth_Age_CPT-City_Epochs_Name](https://user-images.githubusercontent.com/17908951/99893414-bf1f9c80-2c5e-11eb-9200-e64df4943567.png)
![Earth_Age_CPT-City_Geek07](https://user-images.githubusercontent.com/17908951/99893468-6a305600-2c5f-11eb-83a0-518194b57270.png)
![Earth_Age_CPT-City_Geek07_Chron](https://user-images.githubusercontent.com/17908951/99893469-6b618300-2c5f-11eb-9556-055260677f43.png)

**Comment by PaulWessel** [comment]
I agree these are useful and perhaps we can add some of this to the data set description of the [age grids](https://docs.generic-mapping-tools.org/latest/datasets/remote-data.html#global-earth-seafloor-crustal-age-grids)

**Comment by seisman** [comment]
> I agree these are useful and perhaps we can add some of this to the data set description of the [age grids](https://docs.generic-mapping-tools.org/latest/datasets/remote-data.html#global-earth-seafloor-crustal-age-grids)

It sounds good.

What about adding the script by @Esteban82 as an example?

**Comment by Esteban82** [comment]
@seisman I like your idea. 

However I think it may be a bit simpler for an official example. Maybe some other feature/s could be add. What do you think?

**Comment by seisman** [comment]
> However I think it may be a bit simpler for an official example

I actually feel that the official examples are too a little complicated. These official examples are good to show how powerful GMT is and how beautiful plots GMT can produce.

For users, especially new GMT users, they may want to see more "simple" examples to do a simple thing (e.g., plot lines with different styles). 

The [PyGMT Gallery](https://www.pygmt.org/dev/gallery/index.html) provides many "simple" examples, and I think they're more useful.

**Comment by seisman** [comment]
We had some disscussions in https://github.com/GenericMappingTools/gmt/issues/3320 one year ago.

**Comment by seisman** [comment]
@Esteban82 We have already set up the [GMT Examples](https://github.com/GenericMappingTools/gmt-examples) repository to host simple examples. I think your example above is pretty good and simple and should be moved to that repository.

**Comment by Esteban82** [comment]
Thanks seisman. I will keep on mind.

**Comment by joa-quim** [comment]
Where are we in this? Can it be closed>

**Comment by PaulWessel** [comment]
I like the idea of harvesting these examples and add them somewhere - we do have a long to geo types (like me) doing GMT maps like this so try to find a good spot.

**Comment by joa-quim** [comment]
Status of this?

**Comment by Esteban82** [comment]
I think that we could an example. So. this should to documentation. I will add a label.

**Comment by PaulWessel** [comment]
This should probably go in to https://www.generic-mapping-tools.org/remote-datasets/ via the GMT repo called **remote-dataset**s.

**Comment by Esteban82** [comment]
Well in fact I would be better to put it in the [GMT Examples](https://github.com/GenericMappingTools/gmt-examples) repository. So, I think we have to close this issue here and open a new one in that repository.

**Comment by joa-quim** [comment]
We have so many (never used) CPTs that we may as well have these CPTs in our source tree. And than, add these examples.

**Comment by PaulWessel** [comment]
I am fine with ex5? But those CPTs should probably go in the cache

**Comment by joa-quim** [comment]
Can we go along and add those CPTs?

**Comment by joa-quim** [comment]
I mean to say that we should add those CPTs, not closing the issue.

**Comment by Esteban82** [comment]
Yes, sorry.

**Comment by Esteban82** [comment]
So, I have to do the following two things, right?

1. Add the 5 geological CPT to the cache.
2. Add an example in GMT Examples.

**Comment by joa-quim** [comment]
Not sure about **1.** I think we must add them to `share/cpt`. @remkos did the last edits of this, if I'm remembering it right.

**Comment by Esteban82** [comment]
> Not sure about **1.** I think we must add them to `share/cpt`. @remkos did the last edits of this, if I'm remembering it right.

I believe there are two options for adding the CPTs:

1. within GMT (in `share/cpt` ?). Here the CPT will be include in GMT (so we will need a new release).
2. on the GMT [servers](https://github.com/GenericMappingTools/gmtserver-admin/tree/master/cache) as we did for the planetary CPT. No need for a new release. 

I vote for 2.

**Comment by joa-quim** [comment]
But we can do 1. and 2. Where 1. will be official when a new release comes. **2.** is kind of _have it but hidden_

**Comment by remkos** [comment]
I can take care of adding the CPTs to the share/cpt folder. It also requires an addition to the Cookbook where all the included colormaps are shown. I will do that as well. 

It would be great if @Esteban82 could then add the example to GMT examples.

**Comment by joa-quim** [comment]
Thanks, I'll keep a eye and take notes so I'll know how to do it next time.

**Comment by Esteban82** [comment]
Ok @remkos. I will add the example to GMT examples.

**Comment by Esteban82** [comment]
Delayed! Sorry. I will try to do it in July.
